src/bandersnatch/mirror.py

```diff
@@ -290,6 +292,52 @@ def record_finished_package(self, name: str) -> None:
                 ]
                 f.write("\n".join(todo))
 
+    async def cleanup_non_pep_503_paths(self, package: Package) -> None:
+        """
+        Before 4.0 we use to store backwards compatible named dirs for older pip
+        This function checks for them and cleans them up
+        """
+
+        def raw_simple_directory() -> Path:
+            if self.hash_index:
+                return self.webdir / "simple" / package.raw_name[0] / package.raw_name
+            return self.webdir / "simple" / package.raw_name
+
+        def normalized_legacy_simple_directory() -> Path:
+            normalized_name_legacy = utils.bandersnatch_safe_name(package.raw_name)
+            if self.hash_index:
+                return (
+                    self.webdir
+                    / "simple"
+                    / normalized_name_legacy[0]
+                    / normalized_name_legacy
+                )
+            return self.webdir / "simple" / normalized_name_legacy
+
+        if not self.cleanup:
+            return
+
+        logger.debug(f"Running Non PEP503 path cleanup for {package.raw_name}")
+        for deprecated_dir in (
+            raw_simple_directory(),
+            normalized_legacy_simple_directory(),
+        ):
+            # Had to compare path strs as Windows did not match path objects ...
```

Comment:
Hmm - I feel pathlib should handle that :| ... MIght be a new thing due to the storage abstraction from @techalchemy